libs/ledger-live-common/src/bridge/impl.ts

```diff
@@ -25,28 +25,41 @@ export const getCurrencyBridge = (currency: CryptoCurrency): CurrencyBridge => {
     currencyName: currency.name,
   });
 };
+
 export const getAccountBridge = (
   account: AccountLike,
   parentAccount?: Account | null,
 ): AccountBridge<any> => {
   const mainAccount = getMainAccount(account, parentAccount);
   const { currency } = mainAccount;
-  const { family } = currency;
-  const { type } = decodeAccountId(mainAccount.id);
   const supportedError = checkAccountSupported(mainAccount);
 
   if (supportedError) {
     throw supportedError;
   }
 
-  if (type === "mock") {
-    const mockBridge = mockBridges[currency.family];
-    if (mockBridge) return mockBridge.accountBridge;
+  try {
+    return getAccountBridgeByFamily(currency.family, mainAccount.id);
+  } catch {
+    throw new CurrencyNotSupported("currency not supported " + currency.id, {
+      currencyName: currency.name,
+    });
+  }
+};
+
+export function getAccountBridgeByFamily(family: string, accountId?: string): AccountBridge<any> {
+  if (accountId) {
+    const { type } = decodeAccountId(accountId);
+
+    if (type === "mock") {
+      const mockBridge = mockBridges[family];
+      if (mockBridge) return mockBridge.accountBridge;
+    }
   }
 
   const jsBridge = jsBridges[family];
-  if (jsBridge) return jsBridge.accountBridge;
-  throw new CurrencyNotSupported("currency not supported " + currency.id, {
-    currencyName: mainAccount.currency.name,
-  });
-};
+  if (!jsBridge) {
+    throw new CurrencyNotSupported("currency not supported " + family);
```

Comment:
Maybe we could edit the message with "currency bridge not found", to differentiate with supportedCurrencies. As we actually have a setSupportedCurrencies mechanic... We could have a bridge for the currency but it is not supported as we didn't enable it

setSupportedCurrencies will need to get reworked but atleast we can differentiate in the meantime

Reply:
Are you sure about this @hedi-edelbloute ?
I tried to not modify the existing logic. I suppose that if no Bridge is found, the whole currency is not supported at all...

Reply:
Are you aware of setSupportedCurrencies() method ?